x/aswap/msg.go

```diff
@@ -0,0 +1,111 @@
+package aswap
+
+import (
+	"github.com/iov-one/weave"
+	coin "github.com/iov-one/weave/coin"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/migration"
+)
+
+func init() {
+	// Migration needs to be registered for every message introduced in the codec.
+	// This is the convention to message versioning.
+	migration.MustRegister(1, &CreateSwapMsg{}, migration.NoModification)
+	migration.MustRegister(1, &ReleaseSwapMsg{}, migration.NoModification)
+	migration.MustRegister(1, &ReturnSwapMsg{}, migration.NoModification)
+}
+
+const (
+	pathCreateSwap      = "swap/create"
+	pathReleaseSwapMsg  = "swap/release"
+	pathReturnReturnMsg = "swap/return"
+
+	maxMemoSize int = 128
+	// preimage size in bytes
+	preimageSize int = 32
+	// preimageHash size in bytes
+	preimageHashSize int = 32
+)
+
+var _ weave.Msg = (*CreateSwapMsg)(nil)
+var _ weave.Msg = (*ReleaseSwapMsg)(nil)
+var _ weave.Msg = (*ReturnSwapMsg)(nil)
+
+// == ROUTING, Path method fulfills weave.Msg interface to allow routing
+
+func (CreateSwapMsg) Path() string {
+	return pathCreateSwap
+}
+
+func (ReleaseSwapMsg) Path() string {
+	return pathReleaseSwapMsg
+}
+
+func (ReturnSwapMsg) Path() string {
+	return pathReturnReturnMsg
+}
+
+// == VALIDATION, Validate method makes sure basic rules are enforced upon input data and fulfills weave.Msg interface
```

Comment:
pp: we usually don't do this sort of file comments anymore. Can you do godoc instead?